test/bench.cpp

```diff
@@ -1,90 +1,392 @@
+    std::vector<std::string> inputs;
+    std::string type = NULL;
+    std::string output = NULL;
+
+    //coordinates of first and second points
+    float fp_x= 0;
+    float fp_y = 0;
+    float sp_x = 0;
+    float sp_y = 0;
+
+    int coeff_x = 0;
+    int coeff_y = 0;
 
-  /* lines without input and outputs */
-  uint32_t number_of_lines = 0;
+
+    parametrs(const std::vector<std::string> &inputs, const std::string &type, const std::string &output)
+            : inputs(inputs), type(type), output(output) {}
+};
+
+struct bench_statistics {
+    uint32_t number_of_inputs = 0;
+    uint32_t number_of_outputs = 0;
+    uint32_t number_of_dffs = 0;
+
+    /* lines without input and outputs */
+    uint32_t number_of_lines = 0;
+    uint32_t number_of_edges = 0;
+
+    std::vector<parametrs> element;
 };
 
-class bench_statistics_reader : public bench_reader
-{
+
+void addEdge(bench_statistics &stats, const std::vector<std::string> &inputs, const std::string &type = "",
+             const std::string &output = ""){
+    stats.element.emplace_back(inputs, type, output);
+}
+
+void sortGraph(bench_statistics &stats, std::vector<std::vector<parametrs>> &matrix) {
+    std::vector<parametrs> temp;
+
+    for (const auto &iter : stats.element) {
+        std::string type = iter.type;
+        if (type == "input") {
+            temp.emplace_back(iter);
+        }
+    }
+    matrix.emplace_back(temp);
+    temp.clear();
+
+    for (const auto &iter : stats.element) {
+        std::string type = iter.type;
+        if (type != "input" && type != "output") {
+            temp.emplace_back(iter);
+        }
+    }
+
+//  sorting elements and recording to matrix
+    int layer_num = 0;
+        while (temp.size() != 0) {
+            matrix.emplace_back();
+            for (const auto &matrix_it : matrix[layer_num]) {
+                for (int j = 0; j < temp.size(); ++j) {
+                    const auto &temp_inputs = temp[j].inputs;
+                    auto temp_it = std::find(temp_inputs.begin(), temp_inputs.end(), matrix_it.output);
+                    if (temp_it != temp_inputs.end()) {
+                        matrix.back().emplace_back(temp[j]);
+                        temp.erase(temp.begin() + j);
+                        j--;
+                    }
+                }
+            }
+            layer_num++;
+        }
+    temp.clear();
+
+    for (const auto &iter : stats.element) {
+        if (iter.type == "output") {
+            temp.emplace_back(iter);
+        }
+    }
+    matrix.emplace_back(temp);
+    temp.clear();
+}
+
+int maxStr(std::vector<std::vector<parametrs>> &matrix) {
+    size_t max = 0;
+    for (const auto &iter : matrix) {
+        if (max < iter.size()) max = iter.size();
+    }
+    return max;
+}
+
+void calculate_fp_coordinates( std::vector<parametrs>::iterator &it_str,
+                               std::vector<parametrs>::iterator &el_pred_str, std::vector<std::vector<parametrs>> &matrix) {
```

Comment:
Pass iterators by value when not intending to mutate inputs.
Pass `matxi` by const-reference when not intending to mutate.
Indentation size violation